core/src/main/scala/cats/Apply.scala

```diff
@@ -45,6 +37,9 @@ trait Apply[F[_]] extends Functor[F] with ApplyArityFunctions[F] { self =>
       def F: Apply[F] = self
       def G: Apply[G] = GG
     }
+
+  override def product[A, B](fa: F[A], fb: F[B]): F[(A, B)] = ap(fb)(map(fa)(a => b => (a, b)))
```

Comment:
Note that I’m not actually _overriding_ this method, just _implementing_ it, but simulacrum does not accept my code if I omit the `override` qualifier.